### ISSUE TITLE
Discrepancy in NF analysis using Scikit RF

### ISSUE THREAD

**Issue by ghost** [scikit-rf/scikit-rf#538]
I am working on a project that required to extract the noise figure of the whole phased array system including the effect of mismatch connections. The system includes a bunch of active and passive multi-port networks that are cascaded together. The scattering matrix and noisy parameters of each block is also available. 
I am trying to verify the Scikit RF noise figure works fine compare with ADS simulation. In the verification process I found some issues : it seems the scikit RF is not supporting noise figure calculation for the cascade of two multi-port networks. For example, If I connect one output port of the 4-way splitter with known scattering and noisy parameters to an amplifier using Circuit method, It will clean the noisy parameters and the new network comes as the no noise network and there is no way to calculate the noise figure. However, when I use the connect method it will come with the noise parameters but the results is different from ADS. I just wondering when I calculate the noise figure of the 3-way splitter is it included the noisy effect of the other channels? If not the calculated noise figure is the effect of input port noise at the output port 2 or 3? 
As the noise figure analysis of the communication systems has high-demand in industry could you advise how can I cascade the multi-port networks while I can still calculate the overall noise figure and how the network will include the noise effect of other channels on output ?
 
Lets more be specific on this example, I cascaded the 4-way splitter and amplifier with its scattering and noisy parameters using connect method and then take the noise figure of it. The results is not the same as ADS. It seems the overall connection is holding the noise figure of the 4-way splitter and doesn’t include the amplifier NF! Could you please advise if there is an other way to calculated the overall NF of cascaded multi-port networks?
![a1](https://user-images.githubusercontent.com/92495747/137239655-7aef95be-0f4a-45b5-9e7d-1d6778fdc99c.jpg)
![a2](https://user-images.githubusercontent.com/92495747/137239787-c6f9f7dd-829f-4ebc-ac78-2f55f0b37ef0.jpg)

**Comment by jhillairet** [comment]
For sure, `circuit` does not take into account the noise figure data, so it simply delete all noise data.

The cascading of network and associated noise data is still under development in scikit-rf (help welcome!). The cascading of noisy network is partly taken into account, but not very well tested to my knowledge. 

There is for instance, a major work ongoing in PR #401 to create a specific "noisy network" object with additional functionalities for noise aspects.

Could you share some (simplified) data for us to test?

**Comment by mehdisalehi20** [comment]
For sure. I will provide some data based on ADS NF simulation.
BTW, do you have any idea that when it could be developed? We are in a rush to finish this NF analysis in python.

**Comment by jhillairet** [comment]
> For sure. I will provide some data based on ADS NF simulation. 

Thanks.

> BTW, do you have any idea that when it could be developed? We are in a rush to finish this NF analysis in python.

For the when, it also depends on you and your team: scikit-rf is an open-source project, and all contributors are volunteers and not paid for their work. 

Another way to speed up the work is to sponsor it. See the sponsor part (right side) on the [main github page](https://github.com/scikit-rf/scikit-rf) :
![image](https://user-images.githubusercontent.com/4642848/137375925-c28a925a-5bdb-4eb9-b06f-5664bfd298fb.png)

**Comment by mehdisalehi20** [comment]
Here is an example verified in ADS to evaluate the scikit-rf NF tool. It is a combination of active/passive and multi-port networks.
The output of the 4-way splitter is connected to the same amplifiers and then combined again with the same splitter. the NF result from ADS has been attached as well. Can python do the same analysis using Circuit method? Please let me know if you need more configurations?
![Circuit](https://user-images.githubusercontent.com/92495747/137590266-346e207f-0b2e-44dc-9250-260701c4fd44.JPG)
[NF_result.txt](https://github.com/scikit-rf/scikit-rf/files/7357663/NF_result.txt)
[ScateringParameters.zip](https://github.com/scikit-rf/scikit-rf/files/7357664/ScateringParameters.zip)

**Comment by mehdisalehi20** [comment]
Any update or progress on multi-port NF Analysis? any help?

**Comment by jhillairet** [comment]
> Any update or progress on multi-port NF Analysis? any help?

Yes, a big work has recently made by @mbe9a. Please have a look and checkout PR #596 to see if it solves your issue using the traditionnal 'connect' and 'interconnect' methods (not with Circuit).

**Comment by PaulKGrimes** [comment]
I'm trying to look at this, and am getting confused by the implementation and documentation that's merged, and that in #596 , particularly since there appear to be two competing implementations in the Network object.

I'll note that cascading noisy networks, and non-noise source terms (needed for calculating the performance of driven LNAs and mixers) are a solved problem in the Caltech [SuperMix C++ library](http://www.submm.caltech.edu/supermix/), described in their [paper](http://www.submm.caltech.edu/supermix/stt99_supermix.pdf). This is all based on the noise wave correlation matrices of Rutledge & Wedge.

I'm aiming to replicate the functionality of of SuperMix in a Python library, so that we can set up SIS mixer simulations more easily than recompiling C++ code.

**Comment by PaulKGrimes** [comment]
What is the feeling on the two implementations here?

My opinion is that the noise wave method of Rutledge and Wedge is much more well known and general than the noise ABCD matrix method.  I prefer the use of `.c` as the noise correlation matrix parameter, rather than a profusion of `.n_cw`, `.n` etc. which may be conflated with a number of points, in either frequency or ports.
